lib/modules/notifications/index.js

```diff
@@ -291,6 +291,21 @@ export default class Notifications extends ModuleBase {
     return getNativeModule(this).removeDeliveredNotification(notificationId);
   }
 
+  /**
+   * Remove a delivered notifications by tag.
+   * @param tag
+   */
+  removeDeliveredNotificationsByTag(tag: string): Promise<void> {
```

Comment:
Are you able to move this into `AndroidNotifications.js` as it's Android specific functionality?

You'll see it that file, that the JS does a platform check when a method is called and resolves an empty promise if it's not Android.  If we follow that pattern then we don't need to worry about the iOS implementation that you've added.